nix/ci.nix

```diff
@@ -1,39 +1,44 @@
-{ pkgs }:
+{ pkgs, eclint, toolchain }:
 
 let
-  inherit (pkgs) writeScriptBin;
+  inherit (pkgs) writeShellApplication;
 in
-[
+{
 
   # Format
-  (writeScriptBin "ci-check-rustfmt" "cargo fmt --check")
-
-  # Test
-  (writeScriptBin "ci-test-rust" "cargo test")
+  ci-check-rustfmt = (writeShellApplication {
+    name = "ci-check-rustfmt";
+    runtimeInputs = [ toolchain ];
+    text = "cargo fmt --check";
+  });
 
   # Spelling
-  (writeScriptBin "ci-check-spelling" ''
-    codespell \
-      --ignore-words-list ba,sur,crate,pullrequest,pullrequests,ser \
-      --skip target \
-      .
-  '')
+  ci-check-spelling = (writeShellApplication {
+    name = "ci-check-spelling";
+    runtimeInputs = with pkgs; [ codespell ];
+    text = ''
+      codespell \
+        --ignore-words-list ba,sur,crate,pullrequest,pullrequests,ser \
+        --skip target \
+        .
+    '';
+  });
 
   # NixFormatting
-  (writeScriptBin "ci-check-nixpkgs-fmt" ''
-    git ls-files '*.nix' | xargs | nixpkgs-fmt --check
-  '')
+  ci-check-nixpkgs-fmt = (writeShellApplication {
+    name = "ci-check-nixpkgs-fmt";
+    runtimeInputs = with pkgs; [ git ];
```

Comment:
Shouldn't we also put `nixpkgs-fmt` and `findutils` (xargs) in the runtime inputs as well?

Reply:
Yes!

Reply:
I wonder why that didn't fail?

Reply:
Probably because it's already in the devshell, and you use `nix develop` to run the checks. https://github.com/DeterminateSystems/harmonic/pull/88/files#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R74-R76

Reply:
Yup well it's not possible to run a single check in a `nix flake check` :(